### PR TITLE
refactor: allow ES import for cloud string if package type is module

### PR DESCRIPTION
**Pull request by dblythy**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description

Currently, if package type is module, specifying a cloud string does not work.

Related issue: #7559

### Approach

Checks env variable `npm_package_type`, which is autoset, and if it's a `module`, uses `import()` instead of `require()`

### TODOs before merging

- [x] Add test cases
- [x] Add entry to changelog

### CODE REVIEW

**Comment by mtrezza** [comment]
Could you resolve this conflict? Sorry, this will go away with release automation, hopefully by this month.

File: CHANGELOG.md [hunk 1]
```diff
@@ -178,15 +179,15 @@ ___
 
 *Versions >4.5.2 and <4.10.0 are skipped.*
 
-> ⚠️ A security incident caused a number of incorrect version tags to be pushed to the Parse Server repository. These version tags linked to a personal fork of a contributor who had write access to the repository. The code to which these tags linked has not been reviewed or approved by Parse Platform. Even though no releases were published with these incorrect versions, it was possible to define a Parse Server dependency that pointed to these version tags, for example if you defined this dependency:
```
Comment (mtrezza) [inline CHANGELOG.md:181]: What are these changes? I have seen the same ones in a few other PRs as well.

File: spec/cloud/cloudCodeModuleFile.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Parse.Cloud.define('cloudCodeInFile', () => {
```
Comment (mtrezza) [inline spec/cloud/cloudCodeModuleFile.js:1]: Why do we need to add a 3rd file with the same content but a different file name?

Comment (dblythy) [reply spec/cloud/cloudCodeModuleFile.js:1]: The tests fail otherwise

Comment (mtrezza) [reply spec/cloud/cloudCodeModuleFile.js:1]: how so?

Comment (dblythy) [reply spec/cloud/cloudCodeModuleFile.js:1]: `- Unhandled promise rejection: ParseError: 141 Invalid function: "cloudCodeInFile`

**Review by mtrezza — APPROVED** [review]
Looks good!

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/5.0.0-beta.1)

**Comment by Moumouls** [comment]
So as discussed with @mtrezza , here @dblythy  we have an issue, `import` is async. This is a  race issue. A developer do not have any guarantee that the code will be loaded before he try to run some cloud code functions. We have some recurrent CI fails due to this feature (https://github.com/parse-community/parse-server/actions/runs/1426220570)

Last months and also with https://github.com/parse-community/parse-server/pull/7525 we have evidence that the Parse server start up should be refactored with a clean async start function. Then we can drop the old constructor pattern (not flexible at all since we cannot return a promise)

For now the PR could introduce huge problem in a prod environment. i'll recommend to revert this PR.

I'll open an additional Issue with a first suggestion of the spec for the Parse Server start async fn. And also a suggestion of feature implementation to avoid a breaking change with a support of both `constructor` system and `async` fn during one major version.

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0](https://github.com/parse-community/parse-server/releases/tag/5.0.0)

**Pull request merged**